### PR TITLE
 Place outputs in corresponding field to this where inputs were found…

### PR DESCRIPTION
**Pull request by michalkulakowski**
… in request

### CODE REVIEW

File: src/exit_node.hpp [hunk 1]
```diff
@@ -34,12 +34,14 @@ template <typename ResponseType>
 class ExitNode : public Node {
     ResponseType* response;
     const tensor_map_t outputsInfo;
+    bool sharedInputContentsUsed;
 
 public:
-    ExitNode(ResponseType* response, const tensor_map_t& outputsInfo, std::set<std::string> gatherFromNode = {}) :
+    ExitNode(ResponseType* response, const tensor_map_t& outputsInfo, std::set<std::string> gatherFromNode = {}, bool sharedInputContentsUsed = true) :
```
Comment (atobiszei) [inline src/exit_node.hpp:40]: Minor - here we should rename this arg to:
useSharedInputContent -> we tell deserialization where to put the buffer

File: src/serialization.cpp [hunk 2]
```diff
@@ -158,6 +158,63 @@ static void serializeContent(std::string* content, ov::Tensor& tensor) {
...
+            *value = (*(reinterpret_cast<const uint8_t*>((char*)tensor.data() + i)));
+        }
+    } else if (responseOutput.datatype() == "FP64") {
+        for (size_t i = 0; i < tensor.get_byte_size(); i += sizeof(double)) {
+            auto value = responseOutput.mutable_contents()->mutable_fp64_contents()->Add();
+            *value = (*(reinterpret_cast<const double*>((char*)tensor.data() + i)));
+        }
+    } else if (responseOutput.datatype() == "BYTES") {
+        responseOutput.mutable_contents()->add_bytes_contents((char*)tensor.data(), tensor.get_byte_size());
+    }
```
Comment (atobiszei) [inline src/serialization.cpp:215]: Define macro for better readability

File: src/exit_node.hpp (hunk 1 above)
Comment (atobiszei) [inline src/exit_node.hpp:40]: This constructor is not used in pipeline.cpp?

File: src/pipelinedefinition.cpp [hunk 3]
```diff
@@ -228,6 +228,15 @@ Status PipelineDefinition::waitForLoaded(std::unique_ptr<PipelineDefinitionUnloa
     return StatusCode::OK;
 }
 
+static bool useSharedOutputContent(const tensorflow::serving::PredictRequest* request){
+    return true;
+}
+
+static bool useSharedOutputContent(const ::inference::ModelInferRequest* request){
+    return request->raw_input_contents().size() > 0;
+}
```
Comment (atobiszei) [inline src/pipelinedefinition.cpp:237]: Move this out of pipeline definition. Probably to tfs/kfs_utils? Then resuse this function in modelinstance.

File: src/serialization.cpp [hunk 4]
```diff
@@ -158,58 +158,34 @@ static void serializeContent(std::string* content, ov::Tensor& tensor) {
     }
 }
 
+#define serializeByDatatype(contents, datatype)                                            \
```
Comment (atobiszei) [inline src/serialization.cpp]: By convention use capital letters to define macro's
-> #define SERIALIZE_BY_DATATYPE(CONTENTS, DATATYPE)

**Review by atobiszei — APPROVED** [review]

**Review by mzegla — APPROVED** [review]

**Pull request merged**